tests/test_knapsack.py

```diff
@@ -45,13 +45,44 @@ def test_knapsack(self):
         file_path = os.path.join(example_dir, 'main.py')
         data_file_path = os.path.join(example_dir, 'data/small.csv')
         output = subprocess.check_output([sys.executable, file_path,
-                                         data_file_path])
+                                         data_file_path, "50"])
         output = str(output).upper()
         if os.getenv('DEBUG_OUTPUT'):
             print("Example output :-"+ output)
 
         with self.subTest(msg="Verify if output contains 'FOUND SOLUTION' \n"):
             self.assertIn("FOUND SOLUTION", output)
+        with self.subTest(msg="Verify if output contains correct energy' \n"):
+            self.assertIn("ENERGY -205.0", output)
```

Comment:
Is this robustly reproducible? From @joelgdwave's comments sounds like it isn't.

One way to handle the heuristic nature of solver/solution is to retry a few times, if energy doesn't match the expected one.

Another way is to use a less strict assertion, e.g. assert reasonable bounds only.